server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseGetMethod.scala

```diff
@@ -47,20 +47,29 @@ class VacationResponseGetMethod @Inject() (serializer: Serializer,
                        mailboxSession: MailboxSession,
                        processingContext: ProcessingContext): Publisher[Invocation] = {
     metricFactory.decoratePublisherWithTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
-      asVacationResponseGetRequest(invocation.arguments)
-        .flatMap(vacationResponseGetRequest =>
-          getVacationResponse(vacationResponseGetRequest, mailboxSession)
-            .map(vacationResult => VacationResponseGetResponse(
-              accountId = vacationResponseGetRequest.accountId,
-              state = INSTANCE,
-              list = List(vacationResult.vacationResponse),
-              notFound = vacationResult.notFound))
-            .map(vacationResponseGetResponse => Invocation(
-              methodName = methodName,
-              arguments = Arguments(serializer.serialize(vacationResponseGetResponse).as[JsObject]),
-              methodCallId = invocation.methodCallId))))
+      validate(capabilities, invocation)
+        .switchIfEmpty(asVacationResponseGetRequest(invocation.arguments)
+          .flatMap(vacationResponseGetRequest =>
+            getVacationResponse(vacationResponseGetRequest, mailboxSession)
+              .map(vacationResult => VacationResponseGetResponse(
+                accountId = vacationResponseGetRequest.accountId,
+                state = INSTANCE,
+                list = List(vacationResult.vacationResponse),
+                notFound = vacationResult.notFound))
+              .map(vacationResponseGetResponse => Invocation(
+                methodName = methodName,
+                arguments = Arguments(serializer.serialize(vacationResponseGetResponse).as[JsObject]),
+                methodCallId = invocation.methodCallId)))))
+
   }
 
+  private def validate(capabilities: Set[CapabilityIdentifier], invocation: Invocation): SMono[Invocation] =
```

Comment:
Maybe we can use Either for this validation step?

Reply:
Will try